### PR TITLE
RC69: move AddressManager to NodeList thread, fix AccountManager invokes

### PR DESCRIPTION
**Pull request by birarda**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10884/HighFidelity-Beta-PR13463-bb6739c.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10884/HighFidelity-Beta-Interface-PR13463-bb6739c.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10884/HighFidelity-Beta-PR13463-bb6739c.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10884/HighFidelity-Beta-Interface-PR13463-bb6739c.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10896/HighFidelity-Beta-PR13463-f501efd.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10899/HighFidelity-Beta-PR13463-2e9aa82.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10899/HighFidelity-Beta-PR13463-2e9aa82.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10899/HighFidelity-Beta-Interface-PR13463-2e9aa82.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10899/HighFidelity-Beta-PR13463-2e9aa82.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10899/HighFidelity-Beta-Interface-PR13463-2e9aa82.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10909/HighFidelity-Beta-PR13463-c90c891.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10909/HighFidelity-Beta-PR13463-c90c891.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10909/HighFidelity-Beta-Interface-PR13463-c90c891.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10909/HighFidelity-Beta-PR13463-c90c891.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10909/HighFidelity-Beta-Interface-PR13463-c90c891.exe

File: libraries/networking/src/AccountManager.cpp [hunk 1]
```diff
@@ -37,7 +38,7 @@
 #include "UserActivityLogger.h"
 
 
-const bool VERBOSE_HTTP_REQUEST_DEBUGGING = false;
+const bool VERBOSE_HTTP_REQUEST_DEBUGGING = true;
```
Comment (Atlante45) [inline libraries/networking/src/AccountManager.cpp]: Looks like you accidentally committed this.

File: libraries/networking/src/AccountManager.cpp [hunk 2]
```diff
@@ -89,6 +86,10 @@ AccountManager::AccountManager(UserAgentGetter userAgentGetter) :
     qRegisterMetaType<QHttpMultiPart*>("QHttpMultiPart*");
 
     qRegisterMetaType<AccountManagerAuth::Type>();
+
+    connect(this, &AccountManager::callbackProcessed, this, [](QNetworkReply* networkReply) {
+        networkReply->deleteLater();
+    });
```
Comment (Atlante45) [inline libraries/networking/src/AccountManager.cpp]: Do we even need this anymore then? We could just call it in place, no?

File: libraries/networking/src/AccountManager.cpp [hunk 3]
```diff
@@ -323,75 +324,68 @@ void AccountManager::sendRequest(const QString& path,
...
-void AccountManager::passErrorToCallback(QNetworkReply* requestReply) {
-    JSONCallbackParameters callbackParams = _pendingCallbackMap.value(requestReply);
-
-    if (callbackParams.errorCallbackReceiver) {
-        // invoke the right method on the callback receiver
-        QMetaObject::invokeMethod(callbackParams.errorCallbackReceiver, qPrintable(callbackParams.errorCallbackMethod),
-                                  Q_ARG(QNetworkReply&, *requestReply));
+        if (callbackParams.isEmpty()) {
+            connect(networkReply, &QNetworkReply::finished, this, [this]{
+                emit callbackProcessed(qobject_cast<QNetworkReply*>(sender()));
```
Comment (Atlante45) [inline libraries/networking/src/AccountManager.cpp]: If you remove `callbackProcessed`, you could connect directly to `deleteLater`.

File: libraries/networking/src/AccountManager.cpp [hunk 4]
```diff
@@ -323,75 +324,68 @@ void AccountManager::sendRequest(const QString& path,
...
+                }
 
-        // remove the related reply-callback group from the map
-        _pendingCallbackMap.remove(requestReply);
-    } else {
-        if (VERBOSE_HTTP_REQUEST_DEBUGGING) {
-            qCDebug(networking) << "Received error response from metaverse API that has no matching callback.";
-            qCDebug(networking) << "Error" << requestReply->error() << "-" << requestReply->errorString();
-            qCDebug(networking) << requestReply->readAll();
+                emit callbackProcessed(networkReply);
```
Comment (Atlante45) [inline libraries/networking/src/AccountManager.cpp]: We could just call `deleteLater` here now.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10916/HighFidelity-Beta-PR13463-f590511.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10916/HighFidelity-Beta-PR13463-f590511.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10916/HighFidelity-Beta-Interface-PR13463-f590511.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10916/HighFidelity-Beta-PR13463-f590511.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10916/HighFidelity-Beta-Interface-PR13463-f590511.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10919/HighFidelity-Beta-PR13463-f11eb19.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10919/HighFidelity-Beta-Interface-PR13463-f11eb19.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10919/HighFidelity-Beta-PR13463-f11eb19.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10919/HighFidelity-Beta-Interface-PR13463-f11eb19.exe

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
@birarda 
Here are the logs, ready for your approval!

[Logs.zip](https://github.com/highfidelity/hifi/files/2142267/Logs.zip)

**Comment by birarda** [comment]
#### Results

I identified all cases in your log where a change in avatar location occurred because of a domain change. In every case, the domain octree details are cleared prior to the location change. PASS.

```
[06/27 11:29:09] [DEBUG] [default] Called LoadAddress with address  "hifi://hifiqa-rc-bots/4.93274,0.567692,-54.7552/0,-0.0303994,0,0.999538"
[06/27 11:29:10] [DEBUG] [hifi.interface] Clearing domain octree details...
[06/27 11:29:10] [DEBUG] [hifi.interface] MyAvatar goToLocation - moving to 4.93274, 0.567692, -54.7552
[06/27 11:29:10] [DEBUG] [hifi.interface] MyAvatar goToLocation - new orientation is 0, -0.0303994, 0, 0.999538
```

```
[06/27 11:29:17] [DEBUG] [hifi.networking] Trying to go to URL "hifi://hifiqa-rc/501.171,498.496,488.7/0,0.999842,0,-0.0177565"
[06/27 11:29:17] [DEBUG] [hifi.interface] Clearing domain octree details...
[06/27 11:29:17] [DEBUG] [hifi.scriptengine] AssetScriptingInterface::AssetScriptingInterface ScriptEngine(0x1dd1158e960)
[06/27 11:29:17] [DEBUG] [hifi.scriptengine] Clearing ATP scripts from ScriptCache
[06/27 11:29:17] [INFO] [hifi.scriptengine] [about:Entities 5] Script Engine starting:about:Entities 5
[06/27 11:29:17] [DEBUG] [hifi.interface] MyAvatar goToLocation - moving to 501.171, 498.496, 488.7
[06/27 11:29:17] [DEBUG] [hifi.interface] MyAvatar goToLocation - new orientation is 0, 0.999842, 0, -0.0177565
```

```
[06/27 11:29:21] [DEBUG] [hifi.networking] Trying to go to URL "hifi://hifiqa-rc-bots/4.93274,0.567692,-54.7552/0,-0.0303994,0,0.999538"
[06/27 11:29:21] [DEBUG] [hifi.interface] Clearing domain octree details...
[06/27 11:29:21] [DEBUG] [hifi.interface] MyAvatar goToLocation - moving to 4.93274, 0.567692, -54.7552
[06/27 11:29:21] [DEBUG] [hifi.interface] MyAvatar goToLocation - new orientation is 0, -0.0303994, 0, 0.999538
```

```
[06/27 11:29:30] [DEBUG] [hifi.networking] Trying to go to URL "hifi://hifiqa-rc/501.171,498.496,488.7/0,0.999842,0,-0.0177565"
[06/27 11:29:30] [DEBUG] [hifi.interface] Clearing domain octree details...
[06/27 11:29:30] [DEBUG] [hifi.interface] MyAvatar goToLocation - moving to 501.171, 498.496, 488.7
[06/27 11:29:30] [DEBUG] [hifi.interface] MyAvatar goToLocation - new orientation is 0, 0.999842, 0, -0.0177565
```

```
[06/27 11:29:39] [DEBUG] [hifi.networking] Trying to go to URL "hifi://hifiqa-rc-bots/4.93274,0.567692,-54.7552/0,-0.0303994,0,0.999538"
[06/27 11:29:39] [DEBUG] [hifi.interface] Clearing domain octree details...
[06/27 11:29:39] [DEBUG] [hifi.interface] MyAvatar goToLocation - moving to 4.93274, 0.567692, -54.7552
[06/27 11:29:39] [DEBUG] [hifi.interface] MyAvatar goToLocation - new orientation is 0, -0.0303994, 0, 0.999538
```

```
[06/27 11:29:55] [DEBUG] [hifi.networking] Trying to go to URL "hifi://hifiqa-rc/501.171,498.496,488.7/0,0.999842,0,-0.0177565"
[06/27 11:29:56] [DEBUG] [hifi.interface] Clearing domain octree details...
[06/27 11:29:56] [DEBUG] [hifi.interface] MyAvatar goToLocation - moving to 501.171, 498.496, 488.7
[06/27 11:29:56] [DEBUG] [hifi.interface] MyAvatar goToLocation - new orientation is 0, 0.999842, 0, -0.0177565
```

```
[06/27 11:31:57] [DEBUG] [hifi.networking] Trying to go to URL "hifi://hifiqa-rc-bots/4.93274,0.567692,-54.7552/0,-0.0303994,0,0.999538"
[06/27 11:31:57] [DEBUG] [hifi.interface] Clearing domain octree details...
[06/27 11:31:57] [DEBUG] [hifi.interface] MyAvatar goToLocation - moving to 4.93274, 0.567692, -54.7552
[06/27 11:31:57] [DEBUG] [hifi.interface] MyAvatar goToLocation - new orientation is 0, -0.0303994, 0, 0.999538
```

**Pull request merged**